csrc/multidevice/utils.cpp

```diff
@@ -227,6 +227,19 @@ void shardAllLike(TensorView* ref, std::vector<TensorView*> tvs) {
   }
 }
 
+void shardBetween(
+    std::vector<TensorView*> ref_tvs,
+    const std::unordered_set<TensorView*>& boundary_tvs) {
+  auto tvs_between = scheduler_utils::getAllTvsFrom(ref_tvs, boundary_tvs);
+  std::vector<TensorView*> tvs_to_shard;
+  std::copy_if(
+      tvs_between.begin(),
+      tvs_between.end(),
+      std::back_inserter(tvs_to_shard),
+      [](TensorView* tv) { return !tv->hasDeviceMesh(); });
```

Comment:
Use `std::mem_fn` which is more direct. Example: https://github.com/NVIDIA/Fuser/blob/58aa9759ffbe322fa3b9425859a44be27876d8c9/csrc/preseg_passes/propagate_shardings.cpp#L25